src/dmap_import/util_api.py

```diff
@@ -47,23 +47,31 @@ def download_from_url(url: str, local_path: str) -> Optional[str]:
     download_log.log_start()
 
     max_retries = 3
-    for retry_count in range(max_retries):
+    for retry_count in range(max_retries + 1):
         download_log.add_metadata(retry_count=retry_count)
-        with requests.get(url, stream=True, timeout=60 * 5) as response:
-            # retry if status_code not 200
-            if response.status_code != 200:
-                if retry_count == max_retries - 1:
-                    response.raise_for_status()
-                # try again in 15 seconds
-                time.sleep(15)
-                continue
+        try:
+            response = requests.get(url, stream=True, timeout=15)
+            response.raise_for_status()
 
             # download file
             with open(local_path, "wb") as local_file:
                 for file_chunk in response.iter_content(chunk_size=None):
                     # chunk_size=None will write chunks in whatever size they are received
                     local_file.write(file_chunk)
-        break
+
+            response.close()
+            break
+
+        except Exception as exception:
```

Comment:
is there a specific exception that we would expect that would trigger a retry? we could then catch, log, and raise all others.

Reply:
I think we'll need to let this run for a little while a document what kind of Exceptions/Error codes we run into, and then we could modify the retry logic based on that additional information.